_posts/2023/2023-02-13-try-not-to-loop-through-associations-inside-jobs.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Try not to loop through associations inside jobs"
+description: "Your local data isn’t the same as in production"
+layout: article
+category: ruby
+image:
+  base: '2022/try-not-to-loop-through-associations-inside-jobs'
+  alt: "Roller coaster loops"
+  credit: "Priscilla Du Preez"
+  source: "https://unsplash.com/photos/FOsina4f7qM"
+
+---
+
+One of the downside of using an level of abstraction above your database is that it’s easy to accidentally give yourself performance headaches via running multiple very similar queries.
+
+This is typically referred to as an n+1 problem, which is most often seen when you display an Active Record object in a view and show all of the associated records that `belong_to` to it. We often solve this by putting heavier database work in asynchronous jobs.
+
+However, it’s too easy to underestimate how different production data is from your local machine and end up with long running loops _inside_ those jobs.
```

Comment:
*and therefore it's easy to end up with long-running loops...

Reply:
```suggestion
However, it’s too easy to underestimate how different production data is from your local machine and therefore it‘s easy to end up with long running loops _inside_ those jobs.
```